src/pages/iou/IOUModal.js

```diff
@@ -0,0 +1,150 @@
+import React, {Component} from 'react';
+import {View, TouchableOpacity} from 'react-native';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import ONYXKEYS from '../../ONYXKEYS';
+import ROUTES from '../../ROUTES';
+import {redirectToLastReport} from '../../libs/actions/App';
+import IOUAmountPage from './steps/IOUAmountPage';
+import IOUParticipantsPage from './steps/IOUParticipantsPage';
+import IOUConfirmPage from './steps/IOUConfirmPage';
+import Header from '../../components/Header';
+import styles from '../../styles/styles';
+import Icon from '../../components/Icon';
+import {getPreferredCurrency} from '../../libs/actions/IOU';
+import {Close, BackArrow} from '../../components/Icon/Expensicons';
+
+/**
+ * IOU modal for requesting money and splitting bills.
+ */
+const propTypes = {
+    /* Onyx Props */
+
+    // Url currently in view
+    currentURL: PropTypes.string,
+};
+
+// Determines type of step to display within Modal, value provides the title for that page.
+const Steps = {
+    IOUAmount: 'Amount',
+    IOUParticipants: 'Participants',
+    IOUConfirm: 'Confirm',
+};
+
+const defaultProps = {
+    currentURL: '',
+};
+
+class IOUModal extends Component {
+    constructor(props) {
+        super(props);
+
+        this.navigateToPreviousStep = this.navigateToPreviousStep.bind(this);
+        this.navigateToNextStep = this.navigateToNextStep.bind(this);
+
+        this.state = {
+            steps: [Steps.IOUAmount, Steps.IOUParticipants, Steps.IOUConfirm],
+            currentStepIndex: 0,
+            currentStepTitle: Steps.IOUAmount,
+            hasMultipleParticipants: this.props.currentURL === ROUTES.IOU_BILL_SPLIT,
```

Comment:
Hmm I think maybe this component is too general if we are checking this here. It's OK (for the most part) to have an underlying component with various responsibilities - but this shouldn't be checking the url. I'd expect something like this instead 

```jsx
<RightDockedModal route={ROUTES.IOU_SPLIT}>
    <IOUModal hasMultipleParticipants />
</RightDockedModal>
```

Reply:
Cool, this aligns with the other change (removing the route). Made this change.